src/actions/index.js

```diff
@@ -15,3 +17,13 @@ export const isVersionAllowedUpdate = data => ({
 export const apiLoadErrorUpdate = data => ({ type: 'api_load_error_update', payload: data });
 
 export const updateServerInfo = data => ({ type: 'update_server_info', payload: data });
+
+export const toggleTheme = () => {
```

Comment:
suggestion: This action function has side effects even before the actual payload is dispatched.

While this makes sense in this context, it may create confusion and bugs in future maintenances. And it also intrigues me that the action decision is based on the `html` theme, and not on Redux store data.

My initial suggestion would be creating an utils function that implement the side effects while also dispatching the `toggle_theme` action. This would make it clear to the function caller what is the expected result from the get go and avoid context interference.

Do you have other ideas on how to better structure this change?

Reply:
The utility function would require access to the dispatch function, which could complicate the code further. The action currently uses the previous Redux state to toggle the active theme. We rely on the theme string stored in the Redux state for conditional styling in components. However, it's also necessary to set the `data-theme` attribute on the HTML element so that the CSS variables work correctly. We've made some refactorings—please review and see if they address the concerns.